packages/devtools_app/lib/src/charts/chart.dart

```diff
@@ -0,0 +1,588 @@
+        chartController.yCanvasChart + chartController.canvasChartHeight,
+      ),
+      axis,
+    );
+  }
+
+  /// Separated out from drawAxis because we don't know range until plotted.
+  // TODO(terry): Consider combining into drawAxis function.
+  void drawYTicks(Canvas canvas, ChartController chartController, Paint axis) {
+    final xCoord = chartController.xCanvasChart;
+
+    final yScale = chartController.yScale;
+
+    for (var labelIndex = yScale.labelTicks; labelIndex >= 0; labelIndex--) {
+      final unit = pow(10, yScale?.labelUnitExponent).toInt();
+      final y = labelIndex * unit;
+      final yCoord = chartController.yPositonToYCanvasCoord(y);
+
+      final labelName = constructLabel(
+        labelIndex.toInt(),
+        yScale.labelUnitExponent.toInt(),
+      );
+
+      drawText('$labelName', canvas, 5, yCoord);
+
+      // Draw horizontal tick.
+      canvas.drawLine(
+        Offset(xCoord, yCoord),
+        Offset(xCoord - 6, yCoord),
+        axis,
+      );
+    }
+  }
+
+  static String constructLabel(int labelValue, int unitExponent) {
+    var unit = '';
+    switch (unitExponent) {
+      case 1:
+      case 2:
+      case 3:
+        labelValue = labelValue * pow(10, unitExponent);
+        break;
+      // 10K, 100K
+      case 4:
+      case 5:
+        labelValue = labelValue * pow(10, unitExponent - 4);
+        unit = 'K';
+        break;
+      // 1M, 10M, 100M
+      case 6:
+      case 7:
+      case 8:
+        labelValue = labelValue * pow(10, unitExponent - 6);
+        unit = 'M';
+        break;
+      // 1B , 10B, 100B
+      case 9:
+      case 10:
+      case 11:
+        labelValue = labelValue * pow(10, unitExponent - 9);
+        unit = 'B';
+        break;
+      // 1T, 10T, 100T
+      case 12:
+      case 13:
+      case 14:
+        labelValue = labelValue * pow(10, unitExponent - 12);
+        unit = 'T';
+        break;
+    }
+
+    final label = labelValue.toInt();
+    return label == 0 ? '0' : '$label$unit';
+  }
+
+  void drawXTick(
+    Canvas canvas,
+    int timestamp,
+    double xTickCoord,
+    Paint axis, {
+    shortTick = true,
+    displayTime = false,
+  }) {
+    final xAxisYCoord =
+        chartController.yCanvasChart + chartController.canvasChartHeight;
+
+    // Draw vertical tick (short or long).
+    canvas.drawLine(
+      Offset(xTickCoord, xAxisYCoord),
+      Offset(xTickCoord, xAxisYCoord + (shortTick ? 2 : 6)),
+      axis,
+    );
+
+    if (displayTime) {
+      final tp = createText(prettyTimestamp(timestamp), 1);
+      tp.paint(
+        canvas,
+        Offset(xTickCoord - 30, xAxisYCoord + 15 - tp.height / 2),
+      );
+    }
+  }
+
+  void drawText(String key, Canvas canvas, double x, double y) {
+    final tp = createText(key, 1);
+    tp.paint(canvas, Offset(x, y - tp.height / 2));
+  }
+
+  TextPainter createText(String key, double scale) {
+    final span = TextSpan(
+      style: TextStyle(color: Colors.grey[600]),
```

Comment:
does this need to be themeable?

Reply:
Yep, I'll add a TODO only theming is on traces Chart needs characteristics too to pull from the theme.